components/brave_shields/resources/panel/components/tree-list/resource-element.tsx

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import classnames from '$web-common/classnames'
+import style from './resource-element.module.scss'
+import {
+  PermissionButtonHandler
+} from '../../state/component_types'
+
+interface UrlElementProps {
+  path?: string
+  host: string
+  permissionButtonTitle: string
+  onTextExpand?: () => void
+  onPermissionButtonClick: PermissionButtonHandler
+}
+
+function ResourceElement (props: UrlElementProps) {
+  const [isExpanded, setExpanded] = React.useState(false)
+
+  const handleTextClick = () => {
+    // TODO(nullhook): Trigger if the element overflows otherwise we're
+    // recalculating positions.
+    if (isExpanded) return
+    setExpanded(true)
+    props.onTextExpand?.()
+  }
+  const url = props.path ? props.host + props.path : props.host
+  const handlePermissionButtonClick = () => {
+    props.onPermissionButtonClick?.(url)
+
+  }
+
+  const urlTextClass = classnames({
+    [style.textUrl]: true,
+    [style.textMultiline]: isExpanded,
+    [style.textHost]: props.path === undefined
+  })
+
+  const containerClass = classnames({
+    [style.container]: true,
+    [style.containerAlignTop]: isExpanded
+  })
+
+  return (
+    <div className={containerClass}>
+      <div className={urlTextClass} onClick={handleTextClick}>
+        {props.path ? props.path : props.host}
+      </div>
+      {props.onPermissionButtonClick && (
```

Comment:
If onPermissionButtonClick is a required prop, why are we checking if it exists here? If the permission button is conditionally rendering, it would make sense to allow null/undefined values for the prop, which should be consistent throughout the call stack.

Reply:
the handler was suggested in https://github.com/brave/brave-core/pull/17221#discussion_r1128626749

Reply:
Changed type of `onPermissionButtonClick` and made parameter optional